cli/ops.rs

```diff
@@ -755,7 +776,13 @@ fn op_fetch(
           },
         ))
       });
-  Box::new(future)
+  if is_sync {
+    // TODO(afinch7) avoid block_on here?
+    let result_buf = tokio_util::block_on(future)?;
+    Ok(Op::Sync(result_buf))
```

Comment:
Is there a more ideal way to do this? wait?

Reply:
The ideal way would be if the OpCreators themselves returned Op objects. So if they returned a Op::Sync they would be forced to supply the output buf immediately. I have a half-baked patch that edits ops.rs so everyone is returning an Op.

Reply:
Let me rephrase that a little. When we have to use some form of wait I.E. `op_fetch_module_meta_data`, is it better to use `block_on` or just `wait()`?

Reply:
This is `op_fetch` ... I think it should error out if a sync op is attempted.

Ideally `op_fetch_module_meta_data` wouldn't be sync either - but it was the only way I could get the TS compiler to work.